napari/components/viewer_model.py

```diff
@@ -383,17 +383,100 @@ def _sliced_extent_world_augmented(self) -> np.ndarray:
         return self.layers._extent_world_augmented[:, self.dims.displayed]
 
     def reset_view(
-        self, *, margin: float = 0.05, reset_camera_angle: bool = True
+        self, *, margin: float = 0.05, reset_camera_angle: bool | None = None
     ) -> None:
-        """Reset the camera view.
+        """Reset the camera and fit the current layers to the canvas.
+
+        Resets the angles of the camera, adjust the camera zoom,
+        and centers the view so that all layers are visible,
+        accounting for the current grid mode and margin.
 
         Parameters
         ----------
         margin : float in [0, 1)
             Margin as fraction of the canvas, showing blank space around the
-            data.
+            data. Default is 0.05 (5% of the canvas).
+        reset_camera_angle : bool
+            Whether to reset the camera angles to (0, 0, 90) before fitting
+            to view. Default is True.
+            .. deprecated:: 0.6.0
+                The `reset_camera_angle` paramater is deprecated and will be
+                removed in 0.7.0.
+                Continue to use `reset_view()` to reset the camera angles.
+                Use `fit_to_view()` to adjust the camera without
+                resetting the camera angles.
         """
+        if reset_camera_angle is not None:
+            warnings.warn(
+                'reset_camera_angle is deprecated since napari 0.6.0 and will be removed in 0.7.0.'
+                'Continue to use `viewer.reset_view()` to reset the camera angles.'
+                'Use `viewer.fit_to_view()` to adjust the camera without resetting the camera angles.',
+                category=DeprecationWarning,
```

Comment:
```suggestion
                category=FutureWarning,
```
We use `FutureWarning` instead of `DeprecationWarning` as `DeprecationWarning` is silenced by default by python.